clients/tools/bridge/validate_l1_deployment.ts

```diff
@@ -0,0 +1,55 @@
+import { contractsInfo } from "../../config-contracts-info";
+import { extraTokens } from "delphinus-deployment/config/contractsinfo";
+import { getEthConfigs, getConfigByChainName } from "delphinus-deployment/src/config";
+import { L1Client, withL1Client } from "../../client";
+import { L1ClientRole } from "delphinus-deployment/src/types";
+import { encodeL1address } from "web3subscriber/src/addresses";
+
+const bridgeContract = require("../../../build/contracts/Bridge.json");
+
+function crunchTokens() {
+  return contractsInfo.tokens.concat(extraTokens)
+    .filter((x: any) => x.address)
+    .map((x: any) =>
+    encodeL1address(x.address, parseInt(x.chainId).toString(16))
+    );
+}
+
+async function checkL1Deployment(chainName: string, chainId: string) {
+  // Check migrate
+  if(chainId in bridgeContract.networks) {
+    console.log("Truffle migrate to", chainName, "succeed!");
+  } else {
+    console.log("Truffle migrate to", chainName, "failed or was not executed!");
+    process.exit();
+  }
+
+  // Check init.js
+  let ethConfig = await getConfigByChainName(L1ClientRole.Monitor, chainName);
+  try {
+    await withL1Client(ethConfig, false, async (l1client: L1Client) => {
+      let bridge = l1client.getBridgeContract();
+      let existing_tokens = await bridge.allTokens();
+      let init_tokens = crunchTokens();
+      let index = 0;
+      for (let tokenUid of crunchTokens()) {
+        if (existing_tokens[index].token_uid !== tokenUid.toString()) {
+          console.log("Token does not match", existing_tokens[index].token_uid, tokenUid.toString(), "in bridge on", chainName);
+          process.exit();
+        }
+        index++;
+      }
+      console.log("Add token to bridge on", chainName, "succeed!");
+    })
+  } catch (err) {
+    console.log("Add token to bridge on", chainName, "failed: %s", err);
+  }
+}
```

Comment:
Need add checking for delphinus-deployment/config/token-index.json, can see validate_token_status.ts

Reply:
I have added code to check token-index.json.